src/actions/profile-view.js

```diff
@@ -149,6 +150,46 @@ export function selectLeafCallNode(
   };
 }
 
+/**
+ * Given a threadIndex and a sampleIndex, select the call node at the bottom ("root")
+ * of that sample's stack.
+ */
+export function selectRootCallNode(
+  threadIndex: ThreadIndex,
+  sampleIndex: IndexIntoSamplesTable
+): ThunkAction<void> {
+  return (dispatch, getState) => {
+    const threadSelectors = getThreadSelectors(threadIndex);
+    const filteredThread = threadSelectors.getFilteredThread(getState());
+    const callNodeInfo = threadSelectors.getCallNodeInfo(getState());
+
+    const newSelectedStack = filteredThread.samples.stack[sampleIndex];
+    const newSelectedCallNode =
+      newSelectedStack === null
+        ? -1
+        : callNodeInfo.stackIndexToCallNodeIndex[newSelectedStack];
+
+    const rootSelectedCallNode = findRootCallNode(
+      newSelectedCallNode,
+      callNodeInfo
+    );
+    dispatch(
+      changeSelectedCallNode(
+        threadIndex,
+        getCallNodePathFromIndex(
+          rootSelectedCallNode,
+          callNodeInfo.callNodeTable
+        ),
+        // Specify newSelectedCallNode as a descendant call node of rootSelectedCallNode.
+        getCallNodePathFromIndex(
+          newSelectedCallNode,
+          callNodeInfo.callNodeTable
+        )
```

Comment:
strangely, I see it always opens to the "(root)" end. But I think that's OK for now.